### PR TITLE
NMS-9811: Thresholds should work without restart when putting nodes into categories

### PR DESCRIPTION
**Pull request by smith-opennms**
This PR makes Threshd re-evalute filter package maps when a NODE_CATEGORY_MEMBERSHIP_CHANGED_EVENT is received and also when a NODE_GAINED_SERVICE_EVENT_UEI event is received.. Due to the possibly heavy handedness of this, MBean timing metrics for the FilterDao call were added and  exposed.

* JIRA: https://issues.opennms.org/browse/NMS-9811

### CODE REVIEW

File: opennms-config/src/main/java/org/opennms/netmgt/filter/JdbcFilterDao.java [hunk 1]
```diff
@@ -83,6 +87,16 @@
 	private DataSource m_dataSource;
     private DatabaseSchemaConfig m_databaseSchemaConfigFactory;
 
+    private final static MetricRegistry metricRegistry = new MetricRegistry();
+    private final JmxReporter jmxReporter;
+    private final Timer getIpListTimer;
+
+    public JdbcFilterDao() {
+        getIpListTimer = metricRegistry.timer("getIPAddressList");
+        jmxReporter = JmxReporter.forRegistry(metricRegistry).inDomain("org.opennms.netmgt.config").build();
```
Comment (j-white) [inline opennms-config/src/main/java/org/opennms/netmgt/filter/JdbcFilterDao.java]: Let's register the reporter when the bean is initialized (see org.springframework.beans.factory.InitializingBean) and remove it when the bean is destroyed (see org.springframework.beans.factory.DisposableBean) instead if just registering the constructor.

File: opennms-config/src/main/java/org/opennms/netmgt/filter/JdbcFilterDao.java [hunk 2]
```diff
@@ -83,6 +87,16 @@
 	private DataSource m_dataSource;
     private DatabaseSchemaConfig m_databaseSchemaConfigFactory;
 
+    private final static MetricRegistry metricRegistry = new MetricRegistry();
+    private final JmxReporter jmxReporter;
+    private final Timer getIpListTimer;
+
+    public JdbcFilterDao() {
+        getIpListTimer = metricRegistry.timer("getIPAddressList");
```
Comment (j-white) [inline opennms-config/src/main/java/org/opennms/netmgt/filter/JdbcFilterDao.java]: Can we make the MBean domain/name more descriptive. Just by looking at the MBean, it's not clear what org.opennms.netmgt.config#getIPAddressList times.

Comment (smith-opennms) [reply opennms-config/src/main/java/org/opennms/netmgt/filter/JdbcFilterDao.java]: so ... org.opennms.netmgt.config.FilterDao#getIPAddressListForFilter ?
image

Comment (j-white) [reply opennms-config/src/main/java/org/opennms/netmgt/filter/JdbcFilterDao.java]: That helps. I'd keep the FilterDao in the domain portion lower case though :octocat:

**Pull request merged**